06_puppy/mohitnag/syncstore.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+)
+
+// CreatePuppy creates a Puppy in memstore
+func (s *SyncStore) CreatePuppy(p Puppy) error {
+	if _, ok := s.Load(p.ID); !ok {
+		s.Store(p.ID, p)
+		return nil
+	}
+	return fmt.Errorf("puppy with Id %d already exists", p.ID)
+}
+
+// ReadPuppy reads a Puppy from memstore
+func (s *SyncStore) ReadPuppy(id uint32) (Puppy, error) {
+	if puppy, ok := s.Load(id); ok {
+		return puppy.(Puppy), nil
+	}
+	return Puppy{}, fmt.Errorf("puppy with Id %d does not exists", id)
+}
+
+// UpdatePuppy updates a Puppy
+func (s *SyncStore) UpdatePuppy(p Puppy) error {
+	if _, ok := s.Load(p.ID); !ok {
```

Comment:
oops I missed this - this function should be guarded by a Mutex to avoid race conditions as you are accessing your store twice in an atomic fashion. Same for the delete.

Reply:
I think the Locks are part of lab 7. and i have put locks in each operation in there. Can do the same in lab6 as well if you want me to do

Reply:
ok. it was even me who set up this requirement :( (doesn't make sense to me any more).
leave as is.